### PR TITLE
Restaurant Model

### PR DESCRIPTION
**Pull request by YumilRuedaFlores-Ksquare**

### CODE REVIEW

**Review by ArgenisGonzalez-Ksquare — APPROVED** [review]

**Review by ClaudiaDiaz-Ksquare — COMMENTED** [review]
Is the "await Model.sync();" at the end of each model file necessary?

**Review by ArgenisGonzalez-Ksquare — APPROVED** [review]

File: src/models/Restaurant.model.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        description: DataTypes.STRING,
+        city_id: DataTypes.INTEGER,
+        phone_number: DataTypes.STRING,
+        user_id: DataTypes.INTEGER,
+        is_deleted: {
+            type: DataTypes.BOOLEAN,
+            defaultValue: false
+        }
+}, {
+    sequelize // Instance of sequelize that reflects the connection
```
Comment (GabrielOsorno-Ksquare) [inline src/models/Restaurant.model.ts:34]: I think we should all add tableName, so all the table names are standardized at snake_case

**Review by ArgenisGonzalez-Ksquare — DISMISSED** [review]

File: src/routes/Restaurant.routes.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+const router = Router();
+const controller = controllers.restaurantcontrollers;
+/*TODO: MIDDLEWARES*/
+router.get("/:id",controller.getRestaurantById);
+router.get("/:ownerId",controller.getAllRestaurantsbyOwnerId);
+router.post("/new",controller.createRestaurant);
+router.delete("/:id",controller.deleteRestaurantById);
+router.post("/:id",controller.deleteRestaurantById);
+router.post("/:id",controller.updateRestaurantById);
```
Comment (ClaudiaDiaz-Ksquare) [inline src/routes/Restaurant.routes.ts:12]: There are 2 post endpoints to the same route /:id, but they use different controllers, one is for deleting and one for updating apparently. Can you verify which one stays or which one should change of route or endpoint?

**Review by ClaudiaDiaz-Ksquare — CHANGES_REQUESTED** [review]

File: src/models/Restaurant.model.ts (hunk 1 above)
Comment (GabrielOsorno-Ksquare) [reply src/models/Restaurant.model.ts:35]: Would be great if you change it to 'restaurant' C;, same, just for consistency

**Review by ClaudiaDiaz-Ksquare — DISMISSED** [review]
Thanks

File: src/models/Restaurant.model.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        },
+        name: DataTypes.STRING,
+        description: DataTypes.STRING,
+        city_id: DataTypes.INTEGER,
+        phone_number: DataTypes.STRING,
+        user_id: DataTypes.INTEGER,
+        is_deleted: {
+            type: DataTypes.BOOLEAN,
+            defaultValue: false
+        }
```
Comment (GabrielOsorno-Ksquare) [inline src/models/Restaurant.model.ts:32]: You should use allowNull: false on the props that can never be null.

**Review by ClaudiaDiaz-Ksquare — APPROVED** [review]
Thanks

**Review by GabrielOsorno-Ksquare — APPROVED** [review]

**Review by MarianaLopez-Ksquare — DISMISSED** [review]

**Review by GenaroArredondo-Ksquare — APPROVED** [review]

**Review by RodrigoLeon-Ksquare — APPROVED** [review]

**Pull request merged**